### PR TITLE
Add TSX and JSX icons

### PR DESCRIPTION
**Pull request by ashraf7hossain**
Release Notes:

- Added cpp,c-sharp,jsx,tsx icons

**or**

- N/A

Optionally, include screenshots / media showcasing your addition that can be included in the release notes.

### CODE REVIEW

**Comment by ashraf7hossain** [comment]
added some essential icons to the project.

**Review by maxdeviant — CHANGES_REQUESTED** [review]
The C++ and C# icons are too large in comparison with our other icons.

Additionally, all of the icons should be provided with a plain black stroke and full color. The icons themselves should not have colors.

**Comment by Imgkl** [comment]
@ashraf7hossain Please make the icons 14x14, so that it's consistent with other icons.

Also I think you've forgot to add icons path under `types`. You've only added the suffix.

**Comment by ashraf7hossain** [comment]
changed icons size to 14x14 and also added type

**Comment by maxdeviant** [comment]
> changed icons size to 14x14 and also added type

We still need to change the icons to black and white. They still have color.

Also, I think at least the C++ icon (and maybe the C# one) are going to have the same problem I mentioned here https://github.com/zed-industries/zed/pull/8272#issuecomment-1965716583, where the icon doesn't look good in Zed.

**Comment by ashraf7hossain** [comment]
changed color to black and white and added suitable icons.

File: assets/icons/file_icons/cs.svg
Comment (maxdeviant) [inline assets/icons/file_icons/cs.svg:1]: It looks like there's some visual artifacting going on with this icon:

Screenshot 2024-02-29 at 9 05 01 AM

Screenshot 2024-02-29 at 9 05 12 AM

File: assets/icons/file_icons/cpp.svg [hunk 1]
```diff
@@ -0,0 +1,9 @@
...
...4k5HMjx4EVeE5FvBMdCdUa9H3vh/hEX7GlYyIHgj16y4WeeKvEeXjifjZ3MvZ0Z1QGQe5heXRnVAa+SgAMJG7+ER0J1RCB9fyL9GdUCrRhwDdDnA3b/Ix3y6TM/eykJejO6FmcV5mn7awVW77WBr9g1BqeZkBHLWdb9HK3NwcmDSzp7mCB6I7oWa0gF+G7/2au3WyzLs0FedkvheeBM3bXmqYbyeoji1ka3gqNF/rZDljoze9BDCe5XSFp0QztbVcFL3RpZ7m80J4WjRH28ctOTslrAzkfZNv4f9SyN3ruhvPA3yc/0eqtx5KQ/I+HgzfQzZu28x10RtYqmQBG8JTpfHaPm5jdPSmlarRypIcfmu4ftsRVnBG9EaVBuIkltMZnjqN0J7k0uiNKQ3GeazkSHgC1XPbyLXRG1EaipmssAgMqr3MEq+pqBHMZkV4OtVX62CJD1yrkczjH8LTqj7ay3zB5FcjmsM9nhjst63lRp/uUyNrZxlvhSdaHttqFvp+BTWDE1nKlvCEy087yJ3MjN4oUpZGcD2PeX2AV/g6p0ZvDCnGe1nGr8KTMGq/v4rFnu5TsxvGAlZwODwhs2wvcgunRA+8lB/t3Mrz4YmZvm3ndr+yKJU2naU8G56kadqb3MlCL/JJlZzPbWwKT9hap75fVdQQNN8V4nO4iiv5KKOiOzIEz/NDHmI1ndEdUb1rvgJw1GjmsYAFXBjdkQHZy49ZxYNsje6IGkWzFoBu7+Ey5jEv598mfosnWMOjPE1XdFfUWJq9AHQ7lYuZx3zmMDK6Kz1sYzWPs5q1HInuihqTBaC3E/ggs7iAWczkxJAeHOZnrGcd63ma16OHQ43OAlDedGYxi9mcxXTakq7pAJvpYD0bWM8LHIoOXM3DAlCdcUynnelMp51pnEwbYwa9rE52spPX2Pxu62BbdHhqVhaAwRpFG220cTKTGAecyDCGMw4YwwjgLeAQ+4A9HOYAO9nJTnawg93RXZckSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZIkSZJUx/4/lo0uYISbTSAAAAAASUVORK5CYII="/>
```
Comment (maxdeviant) [inline assets/icons/file_icons/cpp.svg:7]: We'll need to change this to be a true SVG.

Right now it looks like the SVG has a PNG embedded inside of it, which isn't what we want.

File: assets/icons/file_icons/jsx.svg [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
...KUgvaoygjfmCQJwWP0ASP2GFE26oiQc5CriCXthNpiSNe8NfCA1r/oIGjp875Xc51E9qp6XCsGsSkHpvdhHZhqnLW/WxcVKi3Av6n24BXlh5qQC5pIG/Tzcit6Vrens3pD2TuyT5zo5ZG+91Y3lsBIuu5pBTtxZbl5nyy4e2UlOTYWSkpb35w4FSlBTnl9wR588xCyZlyYjxAfNqRe8zoiJb6+prb1fX1zSzyqY1TgrxR0YUoLWvxbD+QBOWbdI1GKTHp3iB3yln3UUL09VWBIJ3A9CoGJUMfjQD5817XjJIx7BwG0pqSU4cSYUt/DK6gV7YVpdF8bJ43SIwKfv20CSXRskgFLiGyEKXAXpjbRw3S00S8+TVKYb0nuAbFIAOKjq5eFegGj4e231+MDIqM2eMLrkI5sYxFUbUXrvKHx0g15NMKFsXUfiAUXIcm4QaL4rEWp/eHx0wdm2NA8bTtCwdXoptRyqJIzBfWvaSGx8/vtb21KBLLvr7gWrSxRSgK+uyvIj0oeBKUQVO3G1EUh3uDq1FG5jHoLNZaPDdQC0+M0mfovmYGnUVndAMXFLCpnkVnmMoPj9fAE6aKSC9pRmfQd+brwCV5LzxvQUcx93M3DldDRxC+6oSBQUe1Hn1FBS5KFbPZiA5hbm6Z208JHQQVMjOziEaHVL8TBi7Mb36uBe3WfPr1Yb4K6Eg6D0k53ox2az0y2R1cmrr7gms02oE1560e3FUNHY46YPDqQguLdqCvTvdXgsvTxZ8ymFAIc/3tk4tDoAOLScstb2JRCJPhVLwOnvmGfmLGqQorcrJUFxzLWBCugg5OG71k9/lKE/KoPZsxUQ/P/Jfqubg1+wuaWCRqqTyXtXrmsAAKngqKyFdTPz55qxVtsJT/bcPcCBU88xBV8ODYxb//PL+Gwf8zl5/KXpM86aVIXxU8TZQ+4SPilmzeX2rEhzXe/Pw3SaPD9PAMicYrIKRP30HTk74xc2jfvuG9gvz1KngqURpvv55hfSfPWZb5rV/PT3zx+Z4BHiroQP4NkBiHtt7gSgsAAAAldEVYdGRhdGU6Y3JlYXRlADIwMjEtMDEtMzBUMDI6MDk6MDMrMDA6MDAybV3yAAAAJXRFWHRkYXRlOm1vZGlmeQAyMDIxLTAxLTMwVDAyOjA5OjAzKzAwOjAwQzDlTgAAAABJRU5ErkJggg=="/>
```
Comment (maxdeviant) [inline assets/icons/file_icons/jsx.svg:7]: Same thing here.

We need a true SVG, not and SVG containing a PNG.

**Comment by ashraf7hossain** [comment]
I have added pure svg icons for jsx,cpp and cs and here is the resource 
[link](https://www.figma.com/file/6YOKMaNFlWc7ikJ164UdJI/Untitled?type=design&node-id=0-1&mode=design&t=DJ0rtZnZ9ikleyOV-0)

**Comment by Imgkl** [comment]
@ashraf7hossain Great that you figured out the SVG issue. One more issue I'd like to point out, Look at the icons below as reference. I found your current JSX icon is not similar to what the react framework has. 

image 
image



React - https://www.svgrepo.com/vectors/react/
C++ - https://www.svgrepo.com/svg/521339/c-plusplus-16
c# - https://www.svgrepo.com/svg/521273/c-sharp-16

**Comment by ashraf7hossain** [comment]
Now should I only change react icon or all icons I that I uploaded?

**Comment by ashraf7hossain** [comment]
I have updated react icon please let me know if it works .

**Comment by maxdeviant** [comment]
The C++ and C# icons still don't look good enough to accept. I think we should drop them from this PR and focus on just the TSX and JSX icon.

That one is getting closer, but it still doesn't align with our other icons. It seems like it's almost stretched horizontally.

Screenshot 2024-03-04 at 10 48 17 AM

Where did we source that one from?

**Comment by ashraf7hossain** [comment]
I changed react icon and removed c++ and c# icons , please let me know if it is required something?

**Comment by maxdeviant** [comment]
The icon still doesn't look quite right:

Screenshot 2024-03-14 at 1 17 02 PM

I think I'm going to close this until we can design a TSX/JSX icon for our existing icon theme.

**Pull request closed without merging**